### PR TITLE
Use absolute date range on change page requests

### LINKED ISSUE #6922: Change request when data table page change
| Wazuh | Rev  | Browser                      |
|-------|------|------------------------------|
| 4.9.0 | 05 | Chrome, Firefox, Safari, etc |

**Description**

When the data grid page changes if the filter is relative the date range changes in every page request.
We need to ensure that the date range does not change when we only need a new page to preserve the date range.

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description

This PR modifies the way to get the data grid page documents.
 
Closes #6922

# Tests

## Discover

- Check the `discover` pagination request
> The date range must not change when a data grid table page change

https://github.com/user-attachments/assets/86851977-dc5b-4279-8b20-9703ed82cfe4

- Check the date range changes when the refresh button is clicked or the filter changes

https://github.com/user-attachments/assets/f4088d9f-abaf-4cf4-84cf-02e17589cad2

## Threat hunting dashboard

- Check the `Threat hunting dashboard` data grid pagination request
> The date range must not change when a data grid table page change

https://github.com/user-attachments/assets/f9ec36b2-11ab-463a-90db-5c35c2184fa8

- Check the date range changes when the refresh button is clicked or the filter changes

https://github.com/user-attachments/assets/7d458267-7bda-435f-8dd0-3fda3c890201

## Github and Office Panel

- Check the data grid pagination request
> The date range must not change when a data grid table page change

https://github.com/user-attachments/assets/c8d24947-a1bd-4c33-8a37-09d269157287

- Check the date range changes when the refresh button is clicked or the filter changes

https://github.com/user-attachments/assets/9c0ee502-7ed5-49e1-a7cc-3e9e0ddb69d0

## Dashboards

- Check that the dashboard date range changes when the refresh button is clicked

https://github.com/user-attachments/assets/f60251b8-5d9a-483d-b317-eb56eb942fdc

##  Flyout Discover ( Mitre Framework tab, FIM Inventory Tab, Security Operations Controls tab)

- Check the data grid pagination request
> The date range must not change when a data grid table page change

https://github.com/user-attachments/assets/fd3fb5ad-e614-4e3b-a006-62314633bc7e

- Check the date range changes when the refresh button is clicked or the filter changes

https://github.com/user-attachments/assets/a299bf65-6f71-4f29-b663-5053f99c486c

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by yenienserrano** [comment]
# Tests

## Discover

- Check the `discover` pagination request
> The date range must not change when a data grid table page change



- Check the date range changes when the refresh button is clicked or the filter changes




## Threat hunting dashboard

- Check the `Threat hunting dashboard` data grid pagination request
> The date range must not change when a data grid table page change



- Check the date range changes when the refresh button is clicked or the filter changes



## Github Panel

- Check the data grid pagination request
> The date range must not change when a data grid table page change


- Check the date range changes when the refresh button is clicked or the filter changes



## Dashboards

- Check that the dashboard date range changes when the refresh button is clicked

File: plugins/main/public/components/common/data-source/hooks/use-data-source.ts [hunk 1]
```diff
@@ -11,8 +11,10 @@ import {
   PatternDataSourceFilterManager,
   tFilterManager,
 } from '../index';
+import { TimeRange } from '../../../../../../../src/plugins/data/public';
 import { PinnedAgentManager } from '../../../wz-agent-selector/wz-agent-selector-service';
 import { useIsMounted } from '../../hooks/use-is-mounted';
+import { transformDateRange } from '../../search-bar/search-bar-service';
```
Comment (jbiset) [inline plugins/main/public/components/common/data-source/hooks/use-data-source.ts:17]: `transformDateRange` should not be imported as it is not used. Right?

Comment (Machi3mfl) [reply plugins/main/public/components/common/data-source/hooks/use-data-source.ts]: Solved in https://github.com/wazuh/wazuh-dashboard-plugins/pull/6923/commits/bb1090875952758308c502a73bf2b80e2ad6f1b0

**Review by yenienserrano — APPROVED** [review]
LGTM

**Review by jbiset — APPROVED** [review]
## CR: :green_circle: 

## Test UI: :green_circle: 

<details><summary> 🟢  Discover</summary>

- 🟢 Check the `discover` pagination request
> The date range must not change when a data grid table page change

image


- 🟢 Check the date range changes when the refresh button is clicked or the filter changes

[evidence2.webm](https://github.com/user-attachments/assets/ebbf6073-cd39-4ae8-b2a8-74283406584f)

</details> 

<details><summary> 🟢 Threat hunting dashboard</summary>

- Check the `Threat hunting dashboard` data grid pagination request
> The date range must not change when a data grid table page change

image


- Check the date range changes when the refresh button is clicked or the filter changes

[evidence4.webm](https://github.com/user-attachments/assets/7c499ea8-ebf1-4f98-bf91-165e5555afdc)

</details> 

<details><summary> 🟢 Github Panel</summary>

- Check the data grid pagination request
> The date range must not change when a data grid table page change

image

- Check the date range changes when the refresh button is clicked or the filter changes

[evidence6.webm](https://github.com/user-attachments/assets/0feec6ac-103e-471a-86b4-0caf20f3a762)

</details> 

<details><summary> 🟢  Dashboards</summary>

- Check that the dashboard date range changes when the refresh button is clicked

[evidence7.webm](https://github.com/user-attachments/assets/f7e433b9-687c-4c25-ad21-bc08a3d4762c)

</details> 

<details><summary> 🟢 Flyout Discover ( Mitre Framework tab, FIM Inventory Tab, Security Operations Controls tab)
</summary>

- Check the data grid pagination request
> The date range must not change when a data grid table page change

[evidence8.webm](https://github.com/user-attachments/assets/3c93cdf2-8bd4-48b5-8b74-7c2a9ec38e49)

- Check the date range changes when the refresh button is clicked or the filter changes

[evidence9.webm](https://github.com/user-attachments/assets/52e9de89-892f-4e5f-9f8b-4aaea941967e)

</details>

**Review by asteriscos — APPROVED** [review]
Test :green_circle: 

Peek 2024-08-21 11-16

**Pull request merged**